tetrominos_learning/tetrominos_learning.py

```diff
@@ -0,0 +1,292 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" This script trains a quantum network for encoding Tetris images in the quantum state of two bosonic modes."""
+
+import strawberryfields as sf
+from strawberryfields.ops import Dgate, BSgate, Kgate, Sgate, Rgate
+import tensorflow as tf
+import numpy as np
+import matplotlib.pyplot as plt
+import time
+import os
+
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '2'
+os.environ['OMP_NUM_THREADS'] = '1'
+os.environ['CUDA_VISIBLE_DEVICES'] = '1'
+
+# =============================================
+#   Settings and hyperparameters
+# =============================================
+
+# Model name
+model_string = 'tetris'
+
+# Output folder
+folder_locator = './outputs/'
+
+# Locations of TensorBoard and model saving outputs
+board_string = folder_locator + 'tensorboard/' + model_string + '/'
+save_string = folder_locator + 'models/' + model_string + '/'
+
+
+# Record initial time
+init_time = time.time()
+
+# Set seed for random generator
+tf.set_random_seed(1)
+
+# Depth of the quantum network (suggested: 25)
+depth = 25
+
+# Fock basis truncation
+cutoff = 11  # suggested value: 11
+
+# Image size (im_dim X im_dim)
+im_dim = 4
+
+# Number of optimization steps (suggested: 50000)
+reps = 50000
+
+# Number of steps between data logging/saving.
+partial_reps = 1000
+
+# Number of images to encode (suggested: 7)
+num_images = 7
+
+# Clipping of training parameters
+disp_clip = 5
+sq_clip = 5
+kerr_clip = 1
+
+# Weight for quantum state normalization
+norm_weight = 100.0
+
+# ====================================================
+#   Manual definition of target images
+# ====================================================
+
+train_images = np.zeros((num_images, im_dim, im_dim))
+
+# Target images: L,O,T,I,S,J,Z tetrominos.
+L, O, T, I, S, J, Z = np.zeros((num_images,im_dim, im_dim))
+
+L = np.zeros((im_dim, im_dim))
```

Comment:
Is this line needed given line `82`?

Reply:
no sorry. I removed it with my last commit.